### PR TITLE
[SPARK-45327][BUILD] Upgrade zstd-jni to 1.5.5-7

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to upgrade zstd-jni from 1.5.5-5 to 1.5.5-7.

### Why are the changes needed?
1.Version compare:
- v1.5.5-6 VS v1.5.5-7: https://github.com/luben/zstd-jni/compare/v1.5.5-6...v1.5.5-7
- v1.5.5-5 VS v1.5.5-6: https://github.com/luben/zstd-jni/compare/v.1.5.5-5...v1.5.5-6

2.Some changes include the following:
- Add new method `getFrameContentSize` that will return also the error. https://github.com/luben/zstd-jni/commit/3f6c55e87a38255bdbeeca3e773f99bf9e8a7b7f
- Fix error. https://github.com/luben/zstd-jni/commit/e0c66f1230377d4980ad6abc0767b36c860af538

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
waiting for https://github.com/apache/spark/pull/43345

**Comment by LuciferYang** [comment]
1.5.5-7 released

**Comment by panbingkun** [comment]
> 1.5.5-7 released

Done.

**Review by LuciferYang — APPROVED** [review]
LGTM if test pass

File: core/benchmarks/ZStandardBenchmark-jdk21-results.txt [hunk 1]
```diff
@@ -2,26 +2,26 @@
 Benchmark ZStandardCompressionCodec
 ================================================================================================
 
-OpenJDK 64-Bit Server VM 21+35 on Linux 5.15.0-1046-azure
-Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz
+OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1050-azure
```
Comment (LuciferYang) [inline core/benchmarks/ZStandardBenchmark-jdk21-results.txt:5]: Was the test result generated by 1.5.5-7?

Comment (panbingkun) [reply core/benchmarks/ZStandardBenchmark-jdk21-results.txt:5]: Yes, it's not the result of `1.5.5-6`.
JDK 17: https://github.com/panbingkun/spark/actions/runs/6700489234
JDK 21: https://github.com/panbingkun/spark/actions/runs/6700491773

**Review by dongjoon-hyun — COMMENTED** [review]
+1, late LGTM. Thank you, @panbingkun and @LuciferYang .

**Pull request closed without merging**